pytest-embedded-serial-esp/pytest_embedded_serial_esp/serial.py

```diff
@@ -67,8 +67,9 @@ def __init__(
         self._meta = meta
 
         esptool_target = beta_target or target or 'auto'
-        if port is None:
-            available_ports = esptool.get_port_list()
+        if port is None or port.endswith('*'):
+            port_filter = port.strip('*') if port else ''
+            available_ports = [_p for _p in esptool.get_port_list() if port_filter in _p]
```

Comment:
here should filter out the `ports_to_occupy` as well, since they're "released" by the first-dut's `close`